client/src/components/App/Content/InvestigationForm/TabManagement/ClinicalDetails/useClinicalDetails.ts

```diff
@@ -204,15 +219,45 @@ const useClinicalDetails = (parameters: useClinicalDetailsIncome): useClinicalDe
         );
     };
 
+    const deleteIrrelevantContactEvents = (symptomsStartDate: Date | null, doesHaveSymptoms: boolean) => {
+        const deleteIrrelevantEventsLogger = logger.setup({
+           workflow: 'Deleting irrelevant contact events',
+           investigation: epidemiologyNumber,
+           user: userId
+        });
+        if(Boolean(coronaTestDate)) {
+            const earliestDateToInvestigate = getDatesToInvestigate(doesHaveSymptoms, symptomsStartDate, coronaTestDate)[0];
+            deleteIrrelevantEventsLogger.info('Sending to server date to delete contact events by', Severity.LOW);
+            axios.post('/intersections/deleteContactEventsByDate', {earliestDate: earliestDateToInvestigate}).then((result) => {
+                if(result.data?.data?.deleteContactEventsBeforeDate) {
+                    deleteIrrelevantEventsLogger.info('Deleting contact events finished with success', Severity.LOW);
+                } else {
+                    deleteIrrelevantEventsLogger.error(`Deleting contact events finished with errors: ${result.data}`, Severity.LOW);
+                    alertError('קרתה תקלה במחיקת אירועים מיותרים');
+                }
+            }).catch(err => {
+                deleteIrrelevantEventsLogger.error(`Failed to delete irrelevant contact events: ${err}`, Severity.LOW);
+                alertError('קרתה תקלה במחיקת אירועים מיותרים');
+            })
+        }
+    }
     const saveClinicalDetails = (clinicalDetails: ClinicalDetailsData, validationDate: Date, id: number): void => {
+        didSymptomsDateChangeOccur &&
+            deleteIrrelevantContactEvents(clinicalDetails.symptomsStartDate, clinicalDetails.doesHaveSymptoms);
```

Comment:
You neewdindication whether deleteIrrelevantContactEvents failed for preventing further data savings